fedimint-server/src/consensus/server.rs

```diff
@@ -563,20 +563,18 @@ impl ConsensusServer {
     /// `NUM_EPOCHS_REJOIN_AHEAD` so we can ensure we receive enough HBBFT
     /// messages to produce an outcome.
     async fn rejoin_at_epoch(&mut self, epoch: u64, peer: PeerId) {
-        if let Some(epochs) = self.rejoin_at_epoch.as_mut() {
-            let peers = epochs.entry(epoch).or_default();
-            peers.insert(peer);
-            let threshold = self.cfg.local.p2p_endpoints.threshold();
-
-            if peers.len() >= threshold && self.hbbft.epoch() < epoch {
-                info!(
-                    target: LOG_CONSENSUS,
-                    "Skipping to epoch {}",
-                    epoch + NUM_EPOCHS_REJOIN_AHEAD
-                );
-                self.hbbft.skip_to_epoch(epoch + NUM_EPOCHS_REJOIN_AHEAD);
-                self.request_rejoin(NUM_EPOCHS_REJOIN_AHEAD).await;
-            }
+        let peers = self.rejoin_at_epoch.entry(epoch).or_default();
+        peers.insert(peer);
+        let threshold = self.cfg.local.p2p_endpoints.threshold();
+
+        if peers.len() >= threshold && self.hbbft.epoch() < epoch {
```

Comment:
What do you think about having a threshold here, i.e. only falling in line when lacking behind by a few epochs? Would that create new issues? Otherwise I fear we'll hit this code path quite often?

Reply:
I thought about that, but since we should be receiving messages in-order I don't think we'd hit it often unless messages are dropped.  Since we only hit it after a threshold of peers are ahead, we should already have produced an outcome at that point.